packit/upstream.py

```diff
@@ -445,57 +443,43 @@ def fix_spec(
         )
 
     def _fix_spec_prep(self, archive):
-        prep = self.specfile.spec_content.section("%prep")
-        if not prep:
-            logger.warning("This package doesn't have a %prep section.")
-            return
-
-        # stolen from tito, thanks!
-        # https://github.com/dgoodwin/tito/blob/master/src/tito/common.py#L695
-        regex = re.compile(r"^(\s*%(?:auto)?setup)(.*?)$")
-        for idx, line in enumerate(prep):
-            m = regex.match(line)
-            if m:
-                break
-        else:
-            logger.warning(
-                "This package is not using %(auto)setup macro in prep. "
-                "Packit will not update the %prep section."
-            )
-            return
+        with self.specfile.prep() as prep:
+            if not prep:
+                logger.warning("This package doesn't have a %prep section.")
+                return
+
+            if "%setup" in prep:
+                macro = prep.setup
+            elif "%autosetup" in prep:
+                macro = prep.autosetup
+            else:
+                logger.warning(
+                    "This package is not using %(auto)setup macro in prep. "
+                    "Packit will not update the %prep section."
+                )
+                return
 
-        archive_root_dir = Archive(self, self.get_version()).get_archive_root_dir(
-            archive
-        )
+            archive_root_dir = Archive(self, self.get_version()).get_archive_root_dir(
+                archive
+            )
 
-        new_setup_line = m[1]
-        # replace -n with our -n because it's better
-        args_match = re.search(r"(.*?)\s+-n\s+\S+(.*)", m[2])
-        if args_match:
-            new_setup_line += args_match.group(1)
-            new_setup_line += args_match.group(2)
-        else:
-            new_setup_line += m[2]
-        new_setup_line += f" -n {archive_root_dir}"
-        logger.debug(
-            f"New {'%autosetup' if 'autosetup' in new_setup_line else '%setup'}"
-            f" line:\n{new_setup_line}"
-        )
-        prep[idx] = new_setup_line
-        self.specfile.spec_content.replace_section("%prep", prep)
-        self.specfile.write_spec_content()
+            if "n" in macro.options:
+                # replace -n with our -n because it's better
```

Comment:
Why? :smiley:

Reply:
I don't really know, the comment was already there :slightly_smiling_face: 
But I suppose it's to have the sources extracted to a deterministic path.